client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/ExternalServiceInfo.java

```diff
@@ -0,0 +1,51 @@
+package org.evomaster.client.java.instrumentation;
+
+import java.io.Serializable;
+
+/**
+ * Information related to the external service calls
+ */
+public class ExternalServiceInfo implements Serializable {
+
+    /**
+     * Contains the remote hostname
+     */
+    private final String remoteHostname;
+
+    private final String protocol;
+
+    private final int remotePort;
```

Comment:
what if the port is not specified? maybe this could be a nullable `Integer`?

Reply:
Most case I've observed there is a port. If there is nothing we can use `-1`.